### PR TITLE
2601 appkey

### PR DESCRIPTION
**Pull request by hyi**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]
(1) create a web app tool resource and add extended metadata name-value pair with name as ```appkey``` and value as an application-specific value, say, ```model-my-watershed```.
(2) create a test resource upon which the web app tool created above can act, and add the same extended metadata name-value pair with name as ```appkey``` as shown above.
(3) With matching ```appkey``` name-value pair extended metadata, the web app tool resource automatically shows up in "Open with" list for the test resource as well as on the web app tab. 
This functionality is requested by MMW team and add another granularity of control for users to associate a resource with a specific web app tool resource.

### CODE REVIEW

**Comment by hyi** [comment]
@zhiyuli I assigned you to review this PR since the code is all in web app tool resource implementation. @dtarb asked me to get this functionality implemented and approved as early as I can, so hopefully you can review this PR soon. Thanks.

**Comment by hyi** [comment]
@zhiyuli @dtarb I deployed this branch on playground, so feel free to test it out as needed.

File: hs_tools_resource/app_launch_helper.py [hunk 1]
```diff
@@ -9,39 +9,54 @@ def resource_level_tool_urls(resource_obj, request_obj):
     res_type_str = resource_obj.resource_type
 
     tool_list = []
+    tool_res_id_list = []
     open_with_app_counter = 0
 
+    # associate resources with app tools using extended metadata name-value pair with 'appkey' key
+    tool_app_key = 'appkey'
+    tool_kv_dict = {}
+    for tool_res_obj in ToolResource.objects.all():
```
Comment (pkdash) [inline hs_tools_resource/app_launch_helper.py]: @hyi I think you can simplify this code using the hstore field query api. Have a look at the documentation: https://docs.djangoproject.com/en/1.8/ref/contrib/postgres/fields/#querying-hstorefield

The code needed to find a list of web apps associated with a given resource can be done with only couple of lines of code once you use the hstore specific query api.

**Comment by hydrobuild** [comment]
Build finished.

File: hs_tools_resource/app_launch_helper.py (hunk 1 above)
Comment (hyi) [reply hs_tools_resource/app_launch_helper.py]: @pkdash Thanks for the info. Will update the code to use the hstore field query api.

**Comment by hyi** [comment]
@pkdash Have updated code to use hstorefield query API which indeed simplified code logic quite a bit. Thanks again for the info. Feel free to take a look and let me know if you see any further issues. @zhiyuli It is ready for your code review now that I have addressed @pkdash's comment. I have also updated playground to make sure the same query logic applies. Let me know if you have questions or notice any issues. Thanks.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hydrobuild** [comment]
Build finished.

**Comment by hyi** [comment]
@zhiyuli Can you formally approve this PR so that green check shows up for ```Review required```? Even though you approved it, it is still blocked. So you may need to go through github approve PR UI to formally approve it.

**Pull request merged**